src/storage/memory_store.ts

```diff
@@ -750,7 +770,44 @@ export class MemoryStore implements IStore {
    * @returns {Promise<number>}
    */
   async getGapLimit(): Promise<number> {
-    return this.walletData.gapLimit;
+    if (isGapLimitScanPolicy(this.walletData.scanPolicyData)) {
+      if (!this.walletData.scanPolicyData.gapLimit) {
+        this.walletData.scanPolicyData = {
+          policy: 'gap-limit',
+          gapLimit: GAP_LIMIT,
+        };
+      }
+      return this.walletData.scanPolicyData.gapLimit;
+    }
+    // Return default gap limit
+    return GAP_LIMIT;
```

Comment:
Should we return `null` if scan policy is `index-limit`?

Reply:
The `getGapLimit` of the Storage method will throw if the scan policy is index-limit.
This means the store method can safely return a `number` instead of `number|null`.